### PR TITLE
Sqlsrv identity insert

### PR DESCRIPTION
**Pull request by CyberEkklesiaOwner**
SQLSRV driver does not support ID inserts. 

Proposed solution applicable to sqlsrv only:

https://github.com/laravel/framework/issues/27778#issuecomment-469657265

### CODE REVIEW

File: src/Models/DeletedModel.php [hunk 1]
```diff
@@ -63,6 +67,22 @@ public function restore(Closure $beforeSaving = null): Model
         return $restoredModel;
     }
 
+    protected function enableIdentityInsert(string $modelClass)
+    {
+        if (config('database.default') === 'sqlsrv') {
```
Comment (freekmurze) [inline src/Models/DeletedModel.php:72]: What happens if the model doesn't use the default database?

Comment (CyberEkklesiaOwner) [reply src/Models/DeletedModel.php]: Thank you for the observation. A new commit is summitted.

**Comment by freekmurze** [comment]
To keep the package simple, I'm going to pass on this. You can make it work by specifying a custom model class in the config file 👍

**Comment by CyberEkklesiaOwner** [comment]
No problem. I have a custom in place. I just figured to contribute to make your package compatible with sqlsrv. Anyways, thank you for the opportunity.  The experience helped me to improve my modification. 👍

**Comment by CyberEkklesiaOwner** [comment]
Heads up... On IIS, the config class never worked. The only way that works was the following:

1. CustomDeletedModelServiceProvider class | on ../app/config/app.php
2.  ..app/Http/Providers/CustomDeletedModelServiceProvider.php (to override the binding)
3.  ../app/Http/Models/CustomDeletedModel.php (a copy of your model with my changes). 

If you please reconsider my proposal, your code would be much cleaner for that 'minority' forced to use IIS/Sql Server at workplace.

**Pull request closed without merging**